README.md

```diff
@@ -1,16 +1,1831 @@
+│                                                           └─────────┘           ┆    └─────────┘        │
+│                                                                               With                      │
+│                                                                               Result                    │
+│                                                                  ┌─────────┐   Of         ┌─────────┐   │
+│                                                                  │         │    ┆         │         │   │
+│                                                                  │   Dan   ╞════Set DNS══►│  Erin   │   │
+│                                                                  │         │              │         │   │
+│                                                                  └─────────┘              └─────────┘   │
+│                                                                                                         │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+```
+
+## 1.3 A Note On Serialization
+
+The JSON examples below are given in [DAG-JSON], but UCAN Task is actually defined as IPLD. This makes UCAN Task agnostic to encoding. DAG-JSON follows particular conventions around wrapping CIDs and binary data in tags like so:
+
+### CID
+
+```json
+{"/": "Qmf412jQZiuVUtdgnB36FXFX7xg5V6KEbSJ4dpQuhkLyfD"}
+
+### Bytes
+
+```json
+{"/": {"bytes": "s0m3Byte5"}}
+```
+
+This format help disambiguate type information in generic [DAG-JSON] tooling. However, your presentation need not be in this specific format, as long as it can be converted to and from this cleanly. As it is used for the signature format, [DAG-CBOR] is RECOMMENDED.
+
+## 1.4 Signatures
+
+All payloads described in this spec MUST be signed with a [Varsig].
+
+# 2 High-Level Concepts
+
+## 2.1 Roles
+
+Task adds two new roles to UCAN: invoker and executor. The existing UCAN delegator and delegate principals MUST persist to the invocation.
+
+| UCAN Field | Delegation                             | Task                            |
+| ---------- | -------------------------------------- | ------------------------------- |
+| `iss`      | Delegator: transfer authority (active) | Invoker: request task (active)  |
+| `aud`      | Delegate: gain authority (passive)     | Executor: perform task (active) |
+
+### 2.1.1 Invoker
+
+The invoker signals to the executor that a task associated with a UCAN SHOULD be performed.
+
+The invoker MUST be the UCAN delegator. Their DID MUST be authenticated in the `iss` field of the contained UCAN.
+
+### 2.1.2 Executor
+
+The executor is directed to perform some task described in the UCAN by the invoker.
+
+The executor MUST be the UCAN delegate. Their DID MUST be set the in `aud` field of the contained UCAN.
+
+## 2.2 Components
+
+```mermaid
```

Comment:
I'm unclear what this diagram is trying to show. Is it an example of an Invocation history? At minimum it's probably worth adding some prose to describe this diagram, or move it into an `/examples/histroy.md` file? Thoughts?

Reply:
I tried to show what were the components and how they relate to each other. It was my take this with some more detail, but maybe it was a misinterpretation

![](https://github.com/ucan-wg/invocation/raw/rough/diagrams/concepts.svg)

> At minimum it's probably worth adding some prose to describe this diagram, or move it into an /examples/histroy.md file? Thoughts?

I'm ok with (re)moving this

Reply:
Ah, okay! I wonder if this diagram is mainly too detailed to be on its own. Here's a quick (WIP) sketch of an ERD that's a bit jore abstract:

```mermaid
erDiagram
  Receipt ||--o{ Task : "fork effects"
  Receipt ||--|| Invocation : "join effect"
  Invocation |o--|| Task : contains

  Promise ||--|| Receipt : consume
  Invocation ||--|| Result : produces
  Receipt ||--|| Result : contains
```

The original was meant as a "here's what each depends on other things existing transatively"

```mermaid
flowchart RL
  Invocation --> Task
  Result --> Invocation
  Receipt --> Result
  Promise --> Result
  Memoization -- short-circuit --> Receipt
```

...but perhaps that's not super useful

Reply:
I honest don't find neither my diagram nor above two very informative, so I'd personally remove it.